File: 0680.valid_palindrome_II/solution.py
class Solution:
    def valid_palindrome(self, s: str) -> bool:
        if not s:
            return True
        left, right = 0, len(s) - 1
        while left < right:
            if s[left] == s[right]:
                left += 1
                right -= 1
            else:  # 出现不一致时，去掉左边或右边元素。判断是否是回文
                a = s[left + 1 : right + 1]
                b = s[left:right]
                return a == a[::-1] or b == b[::-1]
        return True

File: 0680.valid_palindrome_II/solution.go
package leetcode0680

func validPalindrome(s string) bool {
	if len(s) == 0 {
		return true
	}
	left, right := 0, len(s)-1
	for left < right {
		if s[left] != s[right] {
			return isPalindrome(s, left+1, right) || isPalindrome(s, left, right-1)
		}
		left++
		right--
	}
	return true
}

func isPalindrome(s string, left, right int) bool {
	for left < right {
		if s[left] != s[right] {
			return false
		}
		left++
		right--
	}
	return true
}
